### PR TITLE
Add browser.secureStorage proposal

### PR DESCRIPTION
**Pull request by oliverdunk**
This adds the proposal from #154 to a new proposals/ folder in the repository. As discussed at the end of the last meeting, my hope is that this makes it easier to collaborate on the proposal while also providing us with a commit history and issue tracker that we can use while moving this forward as a community.

For now, this is very close to a direct port from the shared document. I plan to start making more changes in future PRs.

### CODE REVIEW

File: proposals/secure-storage.md
Comment (Rob--W) [inline proposals/secure-storage.md]: Let's prepend a section "Champions" before the Motivation section, wher the list of people driving the proposal is shown. And include the Github handle, to make it easier for readers/contributors to tag the right person on Github.

Comment (oliverdunk) [reply proposals/secure-storage.md]: I like that! I went ahead and tweaked things as suggested.

File: proposals/secure-storage.md
Comment (dotproto) [inline proposals/secure-storage.md]:
```suggestion
- Oliver Dunk (1Password) - [@oliverdunk](https://github.com/oliverdunk), [email](oliver@1password.com)
```

The downside of the change made in response to @Rob--W's [comment](https://github.com/w3c/webextensions/pull/167#discussion_r812960247) is that we lost your name. Perhaps we could tweak the formatting a bit to keep your name and include your GitHub username. WDYT?

**Review by dotproto — APPROVED** [review]
Bikeshedding in champion name formatting aside, I think this is in a good state to merge & iterate on.

File: proposals/secure-storage.md
Comment (oliverdunk) [reply proposals/secure-storage.md]: I'm happy with that! I accepted your suggestion and tweaked the email, let me know what you think. I can't merge this but I think it's good to go, unless you want anyone else to review.

**Review by mukul-p — APPROVED** [review]

File: proposals/secure-storage.md
Comment (zombie) [inline proposals/secure-storage.md:121]: Do we have a public comment of someone from Dashlane supporting this proposal?  If not, perhaps better to leave this for a later PR they could submit.

**Review by zombie — APPROVED** [review]

File: proposals/secure-storage.md
Comment (zombie) [inline proposals/secure-storage.md:13]: Perhaps a comparison between Apple's and Windows' API surfaces could be useful (later in the proposal).

File: proposals/secure-storage.md
Comment (oliverdunk) [reply proposals/secure-storage.md:121]: @bradcush reached out out-of-band and shared this quote which he was happy for us to use in the proposal. I'm fairly sure he's happy with it being included here but I'll ask if he can drop an approval so we have everything documented in one place.

File: proposals/secure-storage.md
Comment (oliverdunk) [reply proposals/secure-storage.md:13]: That sounds reasonable! I'll resolve this for now but open an issue once we have this merged.

File: proposals/secure-storage.md
Comment (bradcush) [reply proposals/secure-storage.md:121]: This quote is good for us, confirming here publicly on behalf of Dashlane, thanks!

**Pull request merged**
